pipelines/rj_smas__disparo_pic/utils/api_handler.py

```
@@ -0,0 +1,130 @@
+        """Check if token needs refresh based on response status"""
+        if response.status_code == 401:
+            log("Token expired, refreshing...")
+            self._login()
+            return True
+        return False
+
+    def get(self, path: str, params: Optional[Dict] = None, **kwargs) -> requests.Response:
+        """Perform GET request with automatic token refresh"""
+        url = f"{self.base_url}/{path.lstrip('/')}"
+
+        response = requests.get(url, headers=self.headers, params=params, **kwargs)
```

Ruff (0.14.1)

Probable use of `requests` call without timeout

(S113)